Zotero/Scenes/Detail/PDF/Views/CustomFreeTextAnnotationView.swift

```diff
@@ -0,0 +1,186 @@
+                    guard let tagButton else { return }
+                    tagButton.setAttributedTitle(self.attributedString(from: tags), for: .normal)
+                })
+            })
+            .disposed(by: self.disposeBag)
+
+        deleteButton.tintColor = .red
+        deleteButton.rx.tap
+            .observe(on: MainScheduler.instance)
+            .subscribe(with: self, onNext: { [weak deleteButton] `self`, _ in
+                guard let deleteButton else { return }
+                self.delegate?.deleteAnnotation(sender: deleteButton, key: key)
+            })
+            .disposed(by: self.disposeBag)
+
+        let spacer = UIView()
+
+        let container = UIStackView(arrangedSubviews: [sizePicker, spacer, separator, colorButton, separator2, tagButton, separator3, deleteButton])
+        container.translatesAutoresizingMaskIntoConstraints = false
+        container.spacing = 0
+        container.alignment = .center
+        self.addSubview(container)
+
+        NSLayoutConstraint.activate([
+            container.leadingAnchor.constraint(greaterThanOrEqualTo: self.leadingAnchor, constant: 20),
+            self.trailingAnchor.constraint(greaterThanOrEqualTo: container.trailingAnchor, constant: 20),
+            container.centerXAnchor.constraint(equalTo: self.centerXAnchor),
+            self.topAnchor.constraint(equalTo: container.topAnchor),
+            self.bottomAnchor.constraint(equalTo: container.bottomAnchor),
+            separator.widthAnchor.constraint(equalToConstant: 1 / UIScreen.main.scale),
+            separator.heightAnchor.constraint(equalToConstant: 30),
+            separator2.widthAnchor.constraint(equalToConstant: 1 / UIScreen.main.scale),
+            separator2.heightAnchor.constraint(equalToConstant: 30),
+            separator3.widthAnchor.constraint(equalToConstant: 1 / UIScreen.main.scale),
+            separator3.heightAnchor.constraint(equalToConstant: 30),
+            spacer.widthAnchor.constraint(equalToConstant: 20)
+        ])
+    }
+
+    private func attributedString(from tags: [Tag]) -> NSAttributedString {
+        if tags.isEmpty {
+            return NSAttributedString(string: L10n.Pdf.AnnotationsSidebar.addTags, attributes: [.foregroundColor: Asset.Colors.zoteroBlueWithDarkMode.color])
+        } else {
+            return AttributedTagStringGenerator.attributedString(from: tags, limit: 3)
+        }
+    }
+
+    @available(iOS 15, *)
+    private func attributedString(from tags: [Tag]) -> AttributedString {
+        if tags.isEmpty {
+            var string = AttributedString(L10n.Pdf.AnnotationsSidebar.addTags)
+            string.foregroundColor = Asset.Colors.zoteroBlueWithDarkMode.color
+            return string
+        } else {
+            let nsAttributedString = AttributedTagStringGenerator.attributedString(from: tags, limit: 3)
+            return (try? AttributedString(nsAttributedString, including: \.uiKit)) ?? AttributedString()
+        }
+    }
```

Comment:
Is this used?

Reply:
It's used above, where you made your previous comment. But now that we support iOS 15.6+, I can actually get rid of the other `func attributedString(from tags: [Tag]) -> NSAttributedString` and keep this one.

Reply:
Yeah so I thought I could use the `Configuration` API for tag button, but it doesn't support button having just single line of text. So I had to keep classic setup and I removed this as you suggested. e5162c413a6ad621c4e1b0283d49c839b54a4e72